File: src/main/java/ru/openbank/meetup/demo/java/delegation/DelegationUseCase.java
package ru.openbank.meetup.demo.java.delegation;

import java.util.HashMap;
import java.util.Map;

public class DelegationUseCase {

    private static final Map<String, String> map = new HashMap<>();
    private static final String PROMO = "promo";
    private static final String CODE = "code";
    private static final String PHONE = "phone";
    private static final String TYPE = "type";

    public static void main(String[] args) {
        map.put(PROMO, "a value");
        map.put(CODE, "another value");
        map.put(PHONE, "one more value");
        map.put(TYPE, "even more value");

        String promo = map.get(PROMO);
        String code = map.get(CODE);
        String phone = map.get(PHONE);
        String type = map.get(TYPE);
    }
}

File: src/main/kotlin/ru/openbank/meetup/demo/kotlin/delegation/DelegationUseCase.kt
package ru.openbank.meetup.demo.kotlin.delegation

class Application(
    private val mutableMap: MutableMap<String, String?> = HashMap()
) : MutableMap<String, String?> by mutableMap {

    var promo: String? by mutableMap
    var code: String? by mutableMap
    var phone: String? by mutableMap
    var type: String? by mutableMap
}

fun main() {
    val application = Application()

    application.promo = "a value"
    application.code = "another value"
    application.phone = "one more value"
    application.type = "even more value"

    val promo = application.promo
    val code = application.code
    val phone = application.phone
    val type = application.type
}
